test/integration_test/basic_test.go

```diff
@@ -228,25 +241,41 @@ func InstallWithTelemetry(tc *types.TestCase) func(*testing.T) {
 }
 
 func testInstallWithTelemetry(t *testing.T, cluster *corev1.StorageCluster) {
-	secret := testutil.GetExpectedSecret(t, "pure-telemetry-cert.yaml")
-	require.NotNil(t, secret)
+	// Deploy portworx with telemetry set to true
+	cluster, err := ci_utils.CreateStorageCluster(cluster)
+	require.NoError(t, err)
 
-	_, err := coreops.Instance().GetSecret(component.TelemetryCertName, "kube-system")
-	require.True(t, err == nil || errors.IsNotFound(err))
+	err = testutil.ValidateTelemetryInstalled(
+		ci_utils.PxSpecImages,
+		cluster,
+		ci_utils.DefaultValidateDeployTimeout,
+		ci_utils.DefaultValidateDeployRetryInterval)
+	require.NoError(t, err)
 
-	if errors.IsNotFound(err) {
-		secret, err = coreops.Instance().CreateSecret(secret)
-		require.NoError(t, err)
-	} else {
-		secret, err = coreops.Instance().UpdateSecret(secret)
-		require.NoError(t, err)
-	}
+	// Disable telemetry and validate un-installation
+	cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	require.NoError(t, err)
 
-	// Deploy portworx with telemetry set to true
-	cluster, err = ci_utils.CreateStorageCluster(cluster)
+	cluster.Spec.Monitoring.Telemetry.Enabled = false
+	cluster, err = ci_utils.UpdateStorageCluster(cluster)
 	require.NoError(t, err)
 
-	err = testutil.ValidateTelemetry(
+	err = testutil.ValidateTelemetryUninstalled(
```

Comment:
Instead of calling the ValidateInstall and ValidateUninstall, let's just call ValidateTelemetry. Check how it is done for other components. Let's follow a consistent pattern.

Reply:
Actually in the last step after it's uninstalled you can't do ValiateTelemetry, so i explicitly used TelemetryUninstalled and Installed APIs.

Reply:
That makes me think if we want to call ValidateUninstallTelemetry on the last step. Ideally ValidateUninstallStorageCluster should check that px and all components are completely removed. This doesn't apply to just telemetry, but rather to all other components.